### PR TITLE
Add check=false for singular symmetric matrix

### PR DESCRIPTION
**Pull request by alexandrebrilhante**
JuliaLang/LinearAlgebra.jl#630

### CODE REVIEW

**Comment by fredrikekre** [comment]
I believe we need something like:
```diff
diff --git a/stdlib/LinearAlgebra/src/symmetric.jl b/stdlib/LinearAlgebra/src/symmetric.jl
index 14bae24..0fd5cda 100644
--- a/stdlib/LinearAlgebra/src/symmetric.jl
+++ b/stdlib/LinearAlgebra/src/symmetric.jl
@@ -474,18 +474,19 @@ mul!(C::StridedMatrix{T}, A::StridedMatrix{T}, B::Hermitian{T,<:StridedMatrix})
 /(A::Symmetric, x::Number) = Symmetric(A.data/x, sym_uplo(A.uplo))
 /(A::Hermitian, x::Real) = Hermitian(A.data/x, sym_uplo(A.uplo))
 
-function factorize(A::HermOrSym{T}) where T
+factorize(A::HermOrSym) = _factorize(A)
+function _factorize(A::HermOrSym{T}; check::Bool=true) where T
     TT = typeof(sqrt(oneunit(T)))
     if TT <: BlasFloat
-        return bunchkaufman(A)
+        return bunchkaufman(A; check=check)
     else # fallback
-        return lu(A)
+        return lu(A; check=check)
     end
 end
 
-det(A::RealHermSymComplexHerm) = real(det(factorize(A)))
-det(A::Symmetric{<:Real}) = det(factorize(A))
-det(A::Symmetric) = det(factorize(A))
+det(A::RealHermSymComplexHerm) = real(det(_factorize(A; check=false)))
+det(A::Symmetric{<:Real}) = det(_factorize(A; check=false))
+det(A::Symmetric) = det(_factorize(A; check=false))
 
 \(A::HermOrSym{<:Any,<:StridedMatrix}, B::AbstractVector) = \(factorize(A), B)
 # Bunch-Kaufman solves can not utilize BLAS-3 for multiple right hand sides
```
alternatively add `check::Bool` kwarg to all `factorize` methods, and propagate `check=false` only from `det` in this case.

**Comment by andreasnoack** [comment]
The problem was the rounding differences on 32 bit caused the matrix to detected as non-singular but with a tiny pivot so it failed when testing against `0.0`. I've changed the test matrix to one that shouldn't cause rounding differences.

@fredrikekre Why not add the keyword to `factorize`?

**Comment by fredrikekre** [comment]
> @fredrikekre Why not add the keyword to factorize?

We could do that, but then we should probably add it to `factorize(::AbstractMatrix)` also, and then its more than just the minimal bugfix.

**Comment by andreasnoack** [comment]
True. Let's keep it minimal here. Though, it would probably be good to make that change eventually.

File: stdlib/LinearAlgebra/test/symmetric.jl [hunk 1]
```diff
@@ -596,6 +596,14 @@ end
     @test Symmetric(B) == Symmetric(Matrix(B))
 end
 
+@testset "issue #32079: det for singular Symmetric matrix" begin
+    A = ones(3, 3)
+    @test det(Symmetrix(A)) == 0.0
```
Comment (halleysfifthinc) [inline stdlib/LinearAlgebra/test/symmetric.jl]:
```suggestion
    @test det(Symmetric(A)) == 0.0
```

**Review by andreasnoack — APPROVED** [review]

**Pull request merged**